plan.go

```diff
@@ -691,8 +685,8 @@ FieldLoop:
 						return nil, nil, err
 					}
 
-					// add the field to the location
-					fragmentLocations[fieldLocations[0]] = append(fragmentLocations[fieldLocations[0]], field)
+					fieldLocation := p.selectLocation(fieldLocations, config)
```

Comment:
Was this the actual buggy code? The rest of this is simply refactoring so we can use selectLocation, right?

Reply:
Correct, this is the bug and the rest was a refactor to use `selectLocation` as well.